Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m

```diff
@@ -25,81 +26,184 @@
+function [deviceControl, muscleExcitation] = constructControls(x, problem)
+% CONSTRUCTCONTROLS
+%   Construct controls from the optimization optimization variables. Each
+%   variable relates to either:
+%       1) Turning OFF the active device (which releases the loaded spring)  
+%          and turning ON the vastus muscle.
+%       2) Turning ON the active device and turning OFF the vastus muscle
+%          (to reload the spring).
+%   
+%   Each control is described by a 2xN array where the first row contains
+%   time values, the second row contains control values, and N is the 
+%   number of time points:
+%
+%                  control = [t1 t2 ... tN;
+%                             u1 u2 ... uN];
+
+
+% Anonymous functions to conveniently construct 'bang-bang' control signals 
+% for both the active device and the muscle. 
+% 
+% Switch an actuator ON at time 't'. 
+on  =@(t) [t  t+0.01; 
+           0  1.0];
+% Switch an actuator OFF at time 't'.
+off =@(t) [t   t+0.01; 
+           1.0 0];
+       
+% Construct the actuator controls by concatenating a sequence of 'on(t_i)'
+% and 'off(t_i)' anonymous function calls at times t_i = [t1 t2 ... tN].
+% Both problems start with by turning off the muscle and turning on the 
+% active device at t = 0.
+switch problem
+    
+    % The single variable in this problem determines when to simultaneously 
+    % release a loaded passive device (turn OFF the active device) and 
+    % activate the hopper vastus muscle (turn ON the muscle excitation).
+    case 'one_hop'
+        t1 = x;
+        deviceControl    = [on(0)  off(t1)];
+        muscleExcitation = [off(0) on(t1)];
+        
+    % To coordinate the sequence of hops in this problem, the spring must
+    % be loaded and released (with the activated muscle) three times. The
+    % first variable, x(1), is the wait time between when the spring is 
+    % loaded and released for each hop. The x(1) value can be used for all 
+    % three hops since we also optimize the wait time between each hop with 
+    % the other two variables, x(2) and x(3). The active device is turned 
+    % on at t = 0, so no variable is required to load the spring for the 
+    % first hop.
+    case 'three_hops'
+        t1 = x(1);    % Release loaded spring and activate muscle (hop #1) 
+        t2 = x(2)+t1; % Load spring after hop #1
+        t3 = x(1)+t2; % Release loaded spring and activate muscle (hop #2)
+        t4 = x(3)+t3; % Load spring after hop #2
+        t5 = x(1)+t4; % Release loaded spring and activate muscle (hop #3)
+        
+        deviceControl    = [on(0)  off(t1) on(t2)  off(t3) on(t4)  off(t5)];
+        muscleExcitation = [off(0) on(t1)  off(t2) on(t3)  off(t4) on(t5)];    
+end
 
-% Negate peak height for minimization
-f = -peakHeight;
+% For both muscle and device, impose a zero-order hold on the control value
+% to the end of the jump cycle.
+deviceControl    = [deviceControl(1,:) 5;
```

Comment:
I think this is a relic of something I did before, but is there a good place to have a variable such as `finalTime` rather than hard-coding in the `5` in a few places? There's a good chance there isn't, and hence why I was lazy and never did it :-/

Reply:
No, I did this. I don't think there's a need for a variable, but I could clarify that the jump cycle is [0 5] somewhere in the comments.

Reply:
The 5 seconds is hard coded in multiple files.

Reply:
Right, like in the input checking for `BuildInteractiveHopperSolution`. In that case I'll definitely leave it hard coded.